ghost/core/core/server/services/collections/BookshelfCollectionsRepository.js

```diff
@@ -44,28 +46,69 @@ module.exports = class BookshelfCollectionsRepository {
     async getBySlug(slug, options = {}) {
         const model = await this.#model.findOne({slug}, {
             require: false,
-            withRelated: ['collectionPosts'],
             transacting: options.transaction
         });
+
         if (!model) {
             return null;
         }
+
+        model.collectionPostIds = await this.#fetchCollectionPostIds(model.id, options);
+
         return this.#modelToCollection(model);
     }
 
+    /**
+     * NOTE: we are only fetching post_id column here to save memory on
+     *       instances with a large amount of posts
+     *
+     *       The method could be further optimized to fetch posts for
+     *       multiple collections at a time.
+     *
+     * @param {string} collectionId collection to fetch post ids for
+     * @param {Object} options bookshelf options
+     *
+     * @returns {Promise<Array<{post_id: string}>>}
+     */
+    async #fetchCollectionPostIds(collectionId, options = {}) {
+        if (!options.transaction) {
```

Comment:
Why does this _have_ to run in a transaction?

Reply:
Good catch! The method is used across most of the fetch/save methods in the repository. But then It should be up to the calling method to decide if to create/pass in a transaction or not. Would be too much overhead to create a transaction for each getById or getBySlug call

Reply:
have removed this block 👌